### ISSUE TITLE
Add support for pandas 1.4.0

### ISSUE THREAD

**Issue by freddyaboulton** [alteryx/evalml#3275]
Looks like the root cause may be that converting a nullable type to `Categorical` preserves the nullable types in the pandas Categorical categories. On the top is the latest pandas and on the bottom is 1.3.4 pandas.

![image](https://user-images.githubusercontent.com/41651716/150818958-adb14e58-1d51-4632-baad-ff938d238618.png)

This causes np.asarray calls in our imputers to introduce the new pandas null value place-holder, `<NA>` which throws off sklearn:

![image](https://user-images.githubusercontent.com/41651716/150819366-a6246c7a-2ac3-4e67-b31c-6df3c7cae7fb.png)

_Originally posted by @freddyaboulton in https://github.com/alteryx/evalml/issues/3272#issuecomment-1020261986_

**Comment by freddyaboulton** [comment]
Xgboost issue may be fixed here: https://github.com/dmlc/xgboost/issues/7593

**Comment by chukarsten** [comment]
I'm going to have to pick this up to satisfy support for nullable types, @dsherry !  Just FYI, moving to this sprint.

**Comment by thehomebrewnerd** [comment]
@chukarsten Just a heads up, Featuretools doesn't yet fully support `pandas==1.4.0` either. It *might* work for you though, but no guarantees at this point.

https://github.com/alteryx/featuretools/issues/1865

**Comment by chukarsten** [comment]
Ultimately, the problem here is that sklearn can't recognize pd.NA as a type of missing value.  When the imputer looks for a missing value, it generates a mask of the dataframe, where it can recognize None and np.nan, but running the new pandas.NA through that same function returns a simple boolean instead of a similarly shaped array with boolean values in it, with True in the cells for nan values.

I think in the long term, we should submit an issue for sklearn and perhaps a PR to fix this in sklearn.  The function that they have in place for trying to root out NaN values in an array with a dtype of "object" is pretty simple and could use some love/tests. 

```
def _object_dtype_isnan(X):
    return X !=  X
```

Still working on a short term fix that makes sense.

**Comment by chukarsten** [comment]
Looks like the XGBoost warnings are remedied, per https://github.com/dmlc/xgboost/pull/7595!  Just need to wait for a release from XGBoost.
